학기 선택 화면 추가

BEEEAM-J opened the pull request:
## 📌 PR 요약

🌱 작업한 내용
- 학기 선택 화면 추가 하였습니다.

🌱 PR 포인트
- Font 적용하기 전이라 Figma GUI와 약간 상이한 부분이 있는데, 추후 Font 적용 후 수정할 예정입니다.
- Color 또한 임시로 하드코딩 하였습니다. 이 부분 또한 추후 수정할 예정입니다.

## 📸 스크린샷
|스크린샷|
|:--:|
|피그마|
|<img width="744" height="811" alt="image" src="https://github.com/user-attachments/assets/67911cec-47b7-4f78-933d-28aac5bb4733" />|
|https://github.com/user-attachments/assets/7d873e88-ed02-40a9-a125-7ae2794f4896|

## 📮 관련 이슈
- Resolved: #이슈번호

BEEEAM-J committed:
feat: 공용 컴포넌트 추가

- `ChukChukAppBarWithTitle`: 타이틀과 뒤로가기 버튼이 있는 앱 바
- `ChukChukBasicButton`: 기본적인 형태의 버튼
- `ChukChukSelectionContainer`: 선택 가능한 컨테이너

BEEEAM-J committed:
feat: 시간표 생성 - 학기 선택 화면 구현

- 학기 선택 화면 UI를 구현했습니다.
- 선택된 학기에 따라 다음 버튼 활성화 상태를 변경합니다.
- `semesterList`와 `Semester` 데이터 클래스를 `timetableeditor` 패키지에서 `semesterselect` 패키지로 이동했습니다.

BEEEAM-J committed:
feat: 학기 선택 화면 네비게이션 동작 추가

- 시간표 화면에서 시간표 추가 버튼 클릭 시 학기 선택 화면으로 이동하도록 수정

BEEEAM-J committed:
Refactor: `MainActivity.kt` preview to use `App()`

- Changed the preview composable function in `MainActivity.kt` from `SemesterSelectScreen()` to `App()`.

lluke0 commented on composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/common/designsystem/component/button/ChukChukBasicButton.kt:
```diff
@@ -0,0 +1,108 @@
+}
+
+@Composable
+fun BoxScope.ChukChukBasicButton(
```
A: BoxScope로 만드신 이유가 있나요?

BEEEAM-J replied:
버튼을 호출하는 부모 컴포저블이 Box인데, BoxScope로 정의하면 align을 버튼 내에서 정의할 수 있어서 이렇게 처리했습니다. 화면에서 align 정의 하는거보다 더 보기 좋은거 같기도 하다는 생각도 했습니다.

lluke0 replied:
C: BoxScope 이외에서도 ChukChukBasicButton을 호출할 일이 생긴다면 

```
@Composable
fun ChukChukBasicButton(
  modifier: Modifier = Modifier,
  text: String,
  enable: Boolean,
  onClick: () -> Unit,
) {
  val (containerBackground, textColor) = if(enable) {
    Pair(Color(0xff2B2B2B), Color(0xffFFFFFF))
  } else {
    Pair(Color(0xffD9D9E3), Color(0xffA3A2B1))
  }
  val clickableModifier = if (enable) {
    Modifier.clickable(onClick = onClick)
  } else {
    Modifier
  }

  Box(
    modifier = modifier
      .wrapContentHeight()
      .clip(RoundedCornerShape(10.dp))
      .then(clickableModifier)
      .background(containerBackground)
      .padding(18.dp),
  ) {
      Text(
        modifier = Modifier.fillMaxSize(),
        text = text,
        color = textColor,
        fontSize = 18.sp,
        fontWeight = FontWeight.Bold,
        textAlign = TextAlign.Center,
        style = TextStyle(
          letterSpacing = (-0.02).em,
          lineHeight = 1.5.em,
        ),
      )
  }
}
```

요걸 호출하면 되는건가용?

BEEEAM-J replied:
네, 맞습니다.
박스 내에서 버튼 호출하면 자동으로 하단 중앙에 위치 하게 될 것이고, 이 외의 컴포저블에서 호출하면 위치를 임의로 정의해서 사용하면 됩니다. 

학기 선택 화면 외에도 시간표 이름 입력 화면에서도 동일한 위치에 버튼이 위치해야 하는데, 여기에도 사용할 수 있을 것으로 생각이드네요.

lluke0 replied:
R: 음 그러면 Column에서 사용한다면 ColumnScope를 Row면 RowScope를 매번 생성해야할 수도 있을 것 같아요 (이미 BoxScope를 생성했으니)

공통으로 사용되는 디자인 시스템이다보니 특정 상황에 얽매여야하는 코드를 작성하기 보다는 보편적으로 코드를 작성하는건 어떠실까요? BoxScope용 Button을 제거하고 일반 Button에 align을 사용하는게 좋다고 생각합니다!

BEEEAM-J replied:
그럼 BoxScope용 버튼은 제거하고 일반 버튼으로 통합하겠습니다.

lluke0 replied:
넵 반영 후 바로 병합해주시면 될 듯 합니다!

lluke0 commented on composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/presentation/timetable/semesterselect/SemesterSelectScreen.kt:
```diff
@@ -0,0 +1,113 @@
+
+  SuwikiBackground(
+    contentWindowInsets = WindowInsets.navigationBars,
+    color = GrayFB,
```
A: 배경이 기존 수위키 디자인 시스템과 척척학사 디자인 시스템과 같았나용?

BEEEAM-J replied:
디자인시스템을 완전히 확인해보진 않아서 잘 모르겠는데, 저 코드를 추가한 이유는 해당 코드가 없으면 화면이 상태바 위로 넘어가버리는 이슈가 있더라구요.
이 이슈 해결하기 위해서 추가하였습니다.

lluke0 replied:
아 저는 `color = GrayFB,` 요 부분 말씀드린거였어요!

BEEEAM-J replied:
컬러는 정확하게 확인해보지 못했습니다...🥲

lluke0 replied:
아 그렇군요 다음 PR에 반영해주세요!

lluke0 reviewed (approved):
RCA룰로 리뷰하였습니다.
빠르시네요 고생하셨습니다.

R: Request Change -> 반드시 반영해주세요
C: Comment -> 웬만하면 반영해주세요
A: Approve -> 사소한 궁금증

BEEEAM-J committed:
refactor: `ChukChukBasicButton` 높이 고정 및 중복 코드 제거

- `SemesterSelectScreen`에서 `ChukChukBasicButton` 사용 시 `align(Alignment.BottomCenter)`를 추가하여 화면 하단에 버튼을 배치하도록 수정했습니다.

lluke0 commented on composeApp/src/commonMain/kotlin/com/chukchukhaksa/mobile/common/designsystem/component/button/ChukChukBasicButton.kt:
```diff
@@ -44,7 +40,7 @@ fun ChukChukBasicButton(
   Box(
     modifier = modifier
-      .wrapContentHeight()
+      .height(64.dp)
```
아 요거 지금봤네요 fixedHeight보다는 Padding값에 맞추게끔 하는게 사용자가 폰트 사이즈를 커스텀했을 때 디자인이 깨지지않을 것 같아요 @BEEEAM-J

BEEEAM-J replied:
다음 PR에서 수정하도록 하겠습니다.
